src/xrServerEntities/xrServer_Objects_ALife_Items_script.cpp

```diff
@@ -19,6 +19,14 @@ SCRIPT_EXPORT(CSE_ALifeInventoryItem, (),
     [
         class_<CSE_ALifeInventoryItem>("cse_alife_inventory_item")
             //.def(constructor<pcstr>())
+            .def("has_upgrade", +[](CSE_ALifeInventoryItem* ta, pcstr str)
```

Comment:
Looks like definitions are incorrectly defined here.
Has should be checker with return value / add should do mutations.

Reply:
@Neloreck, what do you mean?

Reply:
These function correspond to their origins from CoC. Although, CoC accepts `shared_str` directly, but it's more correct to accept plain C string from scripts.

Reply:
"has_upgrade" script export calls code like
` ta->add_upgrade(str);`

and "add_upgrade" script export calls
`ta->has_upgrade(str);`

So from fist glance checker will add upgrade and `add_upgrade` will just return boolean. Probably it was implemented this way since CoC, but looks like mistake

Reply:
Aaaaaaaaaaaaaaaaaaaaaaaa 😂 
YES, it's my mistake, I mixed them up, this should be vice versa!
Could you submit a quickie fix?

Reply:
https://github.com/OpenXRay/xray-16/pull/1756

Reply:
Thanks!